config/config_file.go

```diff
@@ -43,6 +43,42 @@ func (l KnownConfigLoader) Configure(cfg *Config) error {
 	if !hasExplicitProfile {
 		profile = "DEFAULT"
 	}
+	if cfg.Host != "" && cfg.Profile == "" {
+		err := cfg.fixHostIfNeeded()
+		if err != nil {
+			return err
+		}
+		profiles := []string{}
+		for _, section := range iniFile.Sections() {
+			values := section.KeysHash()
+			host := values["host"]
+			if host == "" {
+				// if host is not set or is empty
+				continue
+			}
+			canonical, err := canonicalHost(host)
+			if err != nil {
+				// we're fine with other corrupt profiles
+				continue
+			}
+			if canonical != cfg.Host {
+				continue
+			}
+			profiles = append(profiles, section.Name())
+		}
+		// in the real situations, we don't expect this to happen often (if not at all),
+		// hence we don't trim the list
+		if len(profiles) > 1 {
+			return fmt.Errorf("%s match %s in %s",
+				strings.Join(profiles, " and "),
+				cfg.Host, configFile)
+		}
```

Comment:
This will error if you auth with `DATABRICKS_HOST` and `DATABRICKS_TOKEN`, and have multiple profiles with that host. Even though the creds would be valid. This path should only be taken as a last resort.

Reply:
good one. will add a testcase for this.

P.S. could you be adding "Request changes" if you want these bigger changes, rather than "comment"? :)

Reply:
> validate: more than one authorization method configured: basic and pat. Config: host=https://dbc-XXXXXXXX-ABCD.cloud.databricks.com, token=***, username=abc, password=***. Env: DATABRICKS_HOST, DATABRICKS_TOKEN

yep, reproduced

Reply:
i wonder what the semantics for `TestConfig_DatabricksCfgProfileResolve_ThroughHost` should be. it picks up `DEFAULT` profile if the default profile is there, but host is different from the one assigned in env.